redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <boost/container/flat_map.hpp>
+#include <utils/json_utils.hpp>
+
+namespace redfish
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            std::string* cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+            if (cableTypeDescription != nullptr)
+            {
+                asyncResp->res.jsonValue["CableType"] = *cableTypeDescription;
+            }
+        }
+        else if (propertyMap.first == "Length")
+        {
+            double* cableLength = std::get_if<double>(&propertyMap.second);
+            if (cableLength != nullptr)
+            {
+                asyncResp->res.jsonValue["LengthMeters"] = *cableLength;
+            }
+        }
+    }
+
+    asyncResp->res.jsonValue["@odata.type"] = "#Cable.v1_0_0.Cable";
+    asyncResp->res.jsonValue["@odata.id"] = "/redfish/v1/Cables/" + cableId;
+    asyncResp->res.jsonValue["Id"] = cableId;
+    asyncResp->res.jsonValue["Name"] = "Cable Id";
+}
+
+/**
+ * The Cable schema
+ */
+inline void requestRoutesCable(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Cables/<str>/")
```

Comment:
How does this handle 
"/redfish/v1/Cables/badCableId/" ?

Reply:
See https://github.com/openbmc/bmcweb/blob/master/COMMON_ERRORS.md#11-not-responding-to-404

Reply:
Or what if you don't have any xyz.openbmc_project.Inventory.Item.Cable and you try to go to this route ?

Reply:
Done